src/map/homunculus.cpp

```diff
@@ -1388,201 +1355,478 @@ uint8 hom_get_intimacy_grade(struct homun_data *hd) {
+						else if (statit.compare("Dex") == 0)
+							hom->gmin.dex = static_cast<uint16>(gbonus);
+						else if (statit.compare("Luk") == 0)
+							hom->gmin.luk = static_cast<uint16>(gbonus);
+					}
+
+					if (this->nodeExists(growth, "Max")) {
+						uint32 gbonus;
+
+						if (!this->asUInt32(growth, "Max", gbonus))
+							return 0;
+
+						if (statit.compare("Hp") == 0)
+							hom->gmax.HP = gbonus;
+						else if (statit.compare("Sp") == 0)
+							hom->gmax.SP = gbonus;
+						else if (statit.compare("Str") == 0)
+							hom->gmax.str = static_cast<uint16>(gbonus);
+						else if (statit.compare("Agi") == 0)
+							hom->gmax.agi = static_cast<uint16>(gbonus);
+						else if (statit.compare("Vit") == 0)
+							hom->gmax.vit = static_cast<uint16>(gbonus);
+						else if (statit.compare("Int") == 0)
+							hom->gmax.int_ = static_cast<uint16>(gbonus);
+						else if (statit.compare("Dex") == 0)
+							hom->gmax.dex = static_cast<uint16>(gbonus);
+						else if (statit.compare("Luk") == 0)
+							hom->gmax.luk = static_cast<uint16>(gbonus);
+					}
+				}
+				
+				if (this->nodeExists(bonus, "EvolutionBonus")) {
+					const YAML::Node &evolution = bonus["EvolutionBonus"];
+
+					if (this->nodeExists(evolution, "Min")) {
+						uint32 ebonus;
+
+						if (!this->asUInt32(evolution, "Min", ebonus))
+							return 0;
+
+						if (statit.compare("Hp") == 0)
+							hom->emin.HP = ebonus;
+						else if (statit.compare("Sp") == 0)
+							hom->emin.SP = ebonus;
+						else if (statit.compare("Str") == 0)
+							hom->emin.str = static_cast<uint16>(ebonus);
+						else if (statit.compare("Agi") == 0)
+							hom->emin.agi = static_cast<uint16>(ebonus);
+						else if (statit.compare("Vit") == 0)
+							hom->emin.vit = static_cast<uint16>(ebonus);
+						else if (statit.compare("Int") == 0)
+							hom->emin.int_ = static_cast<uint16>(ebonus);
+						else if (statit.compare("Dex") == 0)
+							hom->emin.dex = static_cast<uint16>(ebonus);
+						else if (statit.compare("Luk") == 0)
+							hom->emin.luk = static_cast<uint16>(ebonus);
+					}
+
+					if (this->nodeExists(evolution, "Max")) {
+						uint32 ebonus;
+
+						if (!this->asUInt32(evolution, "Max", ebonus))
+							return 0;
+
+						if (statit.compare("Hp") == 0)
+							hom->emax.HP = ebonus;
+						else if (statit.compare("Sp") == 0)
+							hom->emax.SP = ebonus;
+						else if (statit.compare("Str") == 0)
+							hom->emax.str = static_cast<uint16>(ebonus);
+						else if (statit.compare("Agi") == 0)
+							hom->emax.agi = static_cast<uint16>(ebonus);
+						else if (statit.compare("Vit") == 0)
+							hom->emax.vit = static_cast<uint16>(ebonus);
+						else if (statit.compare("Int") == 0)
+							hom->emax.int_ = static_cast<uint16>(ebonus);
+						else if (statit.compare("Dex") == 0)
+							hom->emax.dex = static_cast<uint16>(ebonus);
+						else if (statit.compare("Luk") == 0)
+							hom->emax.luk = static_cast<uint16>(ebonus);
+					}
+				}
```

Comment:
The default values are not required for all the stats ?

Reply:
We also have these labels marked as required during parse.